drivers/sensor/MPU-6050.zig

```diff
@@ -0,0 +1,412 @@
+        y: bool,
+        z: bool,
+    }) Error!void {
+        try self.modify_reg(.accel_config, regs.ACCEL_CONFIG, .{
+            .XA_ST = enable.x,
+            .YA_ST = enable.y,
+            .ZA_ST = enable.z,
+        });
+    }
+
+    pub fn set_gyro_range(self: *MPU_6050, range: GyroRange) Error!void {
+        try self.modify_reg(.gyro_config, regs.GYRO_CONFIG, .{
+            .FS_SEL = range,
+        });
+
+        self.gyro_range = range;
+    }
+
+    pub const GyroRange = enum(u2) {
+        @"250d" = 0,
+        @"500d" = 1,
+        @"1000d" = 2,
+        @"2000d" = 3,
+
+        pub fn sensitivity(range: GyroRange) f32 {
+            return switch (range) {
+                .@"250d" => 131,
+                .@"500d" => 65.5,
+                .@"1000d" => 32.8,
+                .@"2000d" => 16.4,
+            };
+        }
+
+        pub fn scale(range: GyroRange) f32 {
+            return 1.0 / range.sensitivity();
+        }
+    };
+
+    pub fn set_gyro_self_test(self: MPU_6050, enable: packed struct(u3) {
+        x: bool,
+        y: bool,
+        z: bool,
+    }) Error!void {
+        try self.modify_reg(.gyro_config, regs.GYRO_CONFIG, .{
+            .XG_ST = enable.x,
+            .YG_ST = enable.y,
+            .ZG_ST = enable.z,
+        });
+    }
+
+    pub fn read_accel_raw(self: MPU_6050) Error![3]i16 {
+        return try self.read_raw_data(.acc_regx_h, 3);
+    }
+
+    pub fn read_accel(self: MPU_6050) Error![3]f32 {
+        const raw = try self.read_accel_raw();
+        const scale = self.accel_range.scale();
+        return .{
+            @as(f32, @floatFromInt(raw[0])) * scale,
+            @as(f32, @floatFromInt(raw[1])) * scale,
+            @as(f32, @floatFromInt(raw[2])) * scale,
+        };
+    }
+
+    pub fn read_temp_raw(self: MPU_6050) Error!i16 {
+        return try self.read_raw_data(.temp_out_h, 1)[0];
+    }
+
+    pub fn read_temp(self: MPU_6050) Error!f32 {
+        return @as(f32, @floatFromInt(try self.read_temp_raw())) * TEMP_SCALE + TEMP_OFFSET;
+    }
+
+    pub fn read_gyro_raw(self: MPU_6050) Error![3]i16 {
+        return try self.read_raw_data(.gyro_regx_h, 3);
+    }
+
+    pub fn read_gyro(self: MPU_6050) Error![3]f32 {
+        const raw = try self.read_gyro_raw();
+        const scale = self.gyro_range.scale();
+        return .{
+            @as(f32, @floatFromInt(raw[0])) * scale,
+            @as(f32, @floatFromInt(raw[1])) * scale,
+            @as(f32, @floatFromInt(raw[2])) * scale,
+        };
+    }
+
+    pub fn read_accel_temp_gyro_raw(self: MPU_6050) Error!struct {
+        accel: [3]i16,
+        temp: i16,
+        gyro: [3]i16,
+    } {
+        const data = try self.read_raw_data(.acc_regx_h, 7);
+        return .{
+            .accel = .{ data[0], data[1], data[2] },
+            .temp = data[3],
+            .gyro = .{ data[4], data[5], data[6] },
+        };
+    }
+
+    pub fn read_accel_temp_gyro(self: MPU_6050) Error!struct {
+        accel: [3]f32,
```

Comment:
might be nice to split these (could share a type for accel and gyro) so we can access .x .y and .z